cli/pause-plan-data.js

```diff
@@ -0,0 +1,80 @@
+import { ethers } from 'ethers';
+
+const PLOT_TOPIC = "0x46d2fbbb00000000000000000000000000000000000000000000000000000000";
+const EXEC_TOPIC = "0x168ccd6700000000000000000000000000000000000000000000000000000000";
+const DROP_TOPIC = "0x162c7de300000000000000000000000000000000000000000000000000000000";
+
+function decodeLogNote(log, contract) {
+    const eventFragment = contract.interface.getEvent('LogNote');
+    return contract.interface.decodeEventLog(eventFragment, log.data, log.topics).toObject();
+}
+
+function decodeCallParams(sig, fax, contract) {
+    const functionFragment = contract.interface.getFunction(sig);
+    return contract.interface.decodeFunctionData(functionFragment, fax).toObject();
+}
+
+function hash(params) {
+    const abiCoder = new ethers.AbiCoder();
+    const types = ["address", "bytes32", "bytes", "uint256"];
+    const encoded = abiCoder.encode(types, [params.usr, params.tag, params.fax, params.eta]);
+    return ethers.keccak256(encoded);
+}
+
+export async function getFilteredEvents(contract, fromBlock) {
+    try {
+        return await contract.queryFilter([[PLOT_TOPIC, EXEC_TOPIC, DROP_TOPIC]], fromBlock);
+    } catch (error) {
+        console.error("Error fetching filtered events:", error);
+        throw error;
+    }
+}
+
+function processEvent(event, contract) {
+    const decoded = decodeLogNote(event, contract);
+    const decodedCall = decodeCallParams(event.topics[0].slice(0, 10), decoded.fax, contract);
+    return {
+        ...event,
+        decoded,
+        decodedCall,
+        planHash: hash(decodedCall)
+    };
+}
+
+export function prepareData(events, contract, filter) {
```

Comment:
This function conflates data processing with data viewing.
`prepareData` should return an array of objects instead and let the table-formatting to the upper level.

You can and should enrich the data here, (i.e.: figuring out the status), as this part of the data logic, however it should return the data in the most agnostic way possible.